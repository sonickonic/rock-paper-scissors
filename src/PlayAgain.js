import React from "react";
import styled from "styled-components";

const Button = styled.a`
  display: inline-block;
  color: var(--text);
  background-color: var(--white);
  width: 220px;
  padding: 1rem 3.6rem;
  border-radius: 0.7rem;
  margin-top: auto;

  font-weight: 600;
  font-size: 1.6rem;
  letter-spacing: 0.15em;
  text-align: center;
  cursor: pointer;
`;

const Title = styled.p`
  text-align: center;
  font-size: 5.5rem;
  color: var(--white);
  letter-spacing: 0.02em;
  margin-bottom: 1rem;
`;

const Container = styled.div`
  position: absolute;
  bottom: 15rem;
  left: 0;
  right: 0;
  margin-left: auto;
  margin-right: auto;
  text-align: center;
`;

function PlayAgain({ handleClick }) {
  return (
    <Container>
      <Title>you win</Title>
      <Button onClick={handleClick}>play again</Button>
    </Container>
  );
}

export default PlayAgain;
