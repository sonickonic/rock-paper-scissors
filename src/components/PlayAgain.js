import React from "react";
import styled from "styled-components";
import Button from "./Button";

const Title = styled.p`
  text-align: center;
  font-size: 5.5rem;
  color: var(--white);
  letter-spacing: 0.02em;
  margin-bottom: 1rem;
`;

const Container = styled.div`
  position: absolute;
  bottom: 15rem;
  left: 0;
  right: 0;
  margin-left: auto;
  margin-right: auto;
  text-align: center;
`;

const PlayAgain = ({ handleClick, result }) => {
  return (
    <Container>
      <Title>{result === "draw" ? `a ${result}` : `you ${result}`}</Title>
      <Button primary label="play again" handleClick={handleClick} />
    </Container>
  );
};

export default PlayAgain;
