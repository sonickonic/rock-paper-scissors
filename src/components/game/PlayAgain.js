import React from "react";
import styled from "styled-components";
import Button from "../Button";
import { connect } from "react-redux";

const Title = styled.p`
  text-align: center;
  font-size: 5.5rem;
  color: var(--white);
  letter-spacing: 0.02em;
  margin-bottom: 1rem;
`;

const Container = styled.div`
  position: absolute;
  bottom: 15rem;
  left: 0;
  right: 0;
  margin-left: auto;
  margin-right: auto;
  text-align: center;
`;

const PlayAgain = ({ handleClick, result }) => {
  return (
    <Container>
      <Title>{result === "draw" ? `a ${result}` : `you ${result}`}</Title>
      <Button primary label="play again" handleClick={handleClick} />
    </Container>
  );
};

const mapStateToProps = (state) => ({
  result: state.game.result,
});

export default connect(mapStateToProps)(PlayAgain);
