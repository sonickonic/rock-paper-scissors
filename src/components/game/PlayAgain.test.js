import React from "react";
import { render, fireEvent } from "@testing-library/react";
import PlayAgain from "./PlayAgain";

describe("PlayAgain", () => {
  it("renders correctly", () => {
    const result = "win";
    const { asFragment } = render(<PlayAgain result={result} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("shows result text", () => {
    const result = "win";
    const { getByText } = render(<PlayAgain result={result} />);

    expect(getByText("you win")).toBeTruthy();
  });

  it("calls handleClick prop on button", () => {
    const result = "win";
    const handleClick = jest.fn();
    const { container } = render(
      <PlayAgain result={result} handleClick={handleClick} />
    );

    fireEvent.click(container.querySelector("button"));

    expect(handleClick).toBeCalledTimes(1);
  });
});
