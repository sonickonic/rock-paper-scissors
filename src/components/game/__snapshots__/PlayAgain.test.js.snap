// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayAgain renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxiKw cwloZq"
  >
    <p
      class="sc-AxirZ bEbECx"
    >
      you win
    </p>
    <button
      class="sc-AxjAm gvkpfy"
    >
      play again
    </button>
  </div>
</DocumentFragment>
`;
